src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -369,21 +387,21 @@ public Path getJarPath(String jarName) {
   /**
    * Checks whether the Cloud SDK path and version are valid.
    *
-   * @throws CloudSdkNotFoundException when an up-to-date Cloud SDK is not installed where expected
+   * @throws CloudSdkNotFoundException when Cloud SDK is not installed where expected
+   * @throws CloudSdkOutOfDateException when Cloud SDK is out of date
    */
-  public void validateCloudSdk() throws CloudSdkNotFoundException {
+  public void validateCloudSdk() throws CloudSdkNotFoundException, CloudSdkOutOfDateException {
     validateCloudSdkLocation();
     validateCloudSdkVersion();
   }
 
-  private void validateCloudSdkVersion() {
+  private void validateCloudSdkVersion() throws CloudSdkOutOfDateException {
     try {
       CloudSdkVersion version = getVersion();
-      if (version.getMajorVersion() < MINIMUM_VERSION) {
-        throw new CloudSdkOutOfDateException("Cloud SDK version " + version
-            + " is too old. Please update to at least " + MINIMUM_VERSION);
+      if (version.compareTo(MINIMUM_VERSION) < 0) {
+        throw new CloudSdkOutOfDateException(MINIMUM_VERSION);
       }
-    } catch (ProcessRunnerException | IllegalArgumentException ex) {
+    } catch (IllegalArgumentException | IOException ex) {
```

Comment:
Thanks for pointing out. Please take a look at my above [comment](https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/276#discussion_r91149616) for an overall proposal for the exception types to be used in this PR.